designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputText.kt

```diff
@@ -28,12 +31,20 @@ fun InputText(
     state: InputShellState = InputShellState.UNFOCUSED,
     supportingText: List<SupportingTextData>? = null,
     legendData: LegendData? = null,
-    inputText: String = "",
+    inputText: String? = null,
     isRequiredField: Boolean = false,
+    onNextClicked: (() -> Unit)? = null,
+    onValueChanged: ((String?) -> Unit)? = null,
+    isLastField: Boolean = false,
     modifier: Modifier = Modifier,
 ) {
-    var inputValue by rememberSaveable { mutableStateOf(inputText) }
-    var deleteButtonIsVisible by remember { mutableStateOf(inputText.isNotEmpty()) }
+    val inputValue by remember(inputText) { mutableStateOf(inputText) }
+
+    var deleteButtonIsVisible by remember { mutableStateOf(!inputText.isNullOrEmpty()) }
+    val focusManager = LocalFocusManager.current
+
+    val keyboardOptions by remember(isLastField) { if (isLastField) { mutableStateOf(KeyboardOptions(imeAction = ImeAction.Done)) } else { mutableStateOf(KeyboardOptions(imeAction = ImeAction.Next)) } }
```

Comment:
why this logic is in the view component?, shouldn't be presentation logic?, maybe the component can receive the KeyboardOption as parameter or the ImeAction type. What do you think?

Reply:
yes, good idea